### PR TITLE
Enable ComponentSLI as beta feature

### PR DESCRIPTION
**Pull request by richabanker**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Enables ComponentSLIs as beta feature
Adds an e2e test to verify that health metrics are available at API server's /metrics/slis endpoint

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
`/metrics/slis` is made available for control plane components allowing you to scrape health check metrics.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://kep.k8s.io/3466
```

### CODE REVIEW

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

File: test/e2e/framework/metrics/metrics_grabber.go [hunk 1]
```diff
@@ -383,6 +393,14 @@ func (g *Grabber) Grab(ctx context.Context) (Collection, error) {
 			}
 		}
 	}
+	if g.grabFromAPIServer {
```
Comment (SergeyKanzhelev) [inline test/e2e/framework/metrics/metrics_grabber.go:396]: the same if condition as on line 340. Expected?

Comment (logicalhan) [reply test/e2e/framework/metrics/metrics_grabber.go:396]: ah yeah, we should move this block up to 340

Comment (Richabanker) [reply test/e2e/framework/metrics/metrics_grabber.go]: Moved to the single if block on line 340. But some checks are currently failing..

**Review by logicalhan — COMMENTED** [review]
/lgtm
/approve

File: test/e2e/framework/metrics/metrics_grabber.go [hunk 2]
```diff
@@ -334,6 +344,12 @@ func (g *Grabber) Grab(ctx context.Context) (Collection, error) {
 		} else {
 			result.APIServerMetrics = metrics
 		}
+		metrics, err = g.GrabMetricsSLIsFromAPIServer(ctx)
+		if err != nil {
+			errs = append(errs, err)
```
Comment (SergeyKanzhelev) [inline test/e2e/framework/metrics/metrics_grabber.go:349]: so tests will fail when tested with the feature gate turned off? It looks like a common pattern, so this is not the comment to this PR. Just in general, not an ideal practice

/lgtm

File: test/e2e/framework/metrics/api_server_metrics.go [hunk 3]
```diff
@@ -50,3 +50,11 @@ func (g *Grabber) getMetricsFromAPIServer(ctx context.Context) (string, error) {
 	}
 	return string(rawOutput), nil
 }
+
+func (g *Grabber) getMetricsSLIsFromAPIServer(ctx context.Context) (string, error) {
```
Comment (lavalamp) [inline test/e2e/framework/metrics/api_server_metrics.go:54]: Why are `Grabber` methods not all in the same file?

File: test/e2e/framework/metrics/metrics_grabber.go (hunk 2 above)
Comment (Richabanker) [reply test/e2e/framework/metrics/metrics_grabber.go:390]: feature gate should be set to True for beta features, which is what we are doing in this PR and hence have added this test. But if the feature was still alpha, we wouldn't have a test for it.   @logicalhan to confirm this.

File: test/e2e/framework/metrics/api_server_metrics.go (hunk 3 above)
Comment (Richabanker) [reply test/e2e/framework/metrics/api_server_metrics.go]: Yeah that is strange, thanks for bringing this to notice. Moved all Grabber methods to metrics_grabber.

File: test/e2e/framework/metrics/metrics_grabber.go (hunk 2 above)
Comment (logicalhan) [reply test/e2e/framework/metrics/metrics_grabber.go:390]: Yeah, there isn't an API we can hit in order to determine feature enablement, so we can't actually toggle tests depending on enablement. Therefore we end up with this pattern.

**Pull request merged**